waves/project.py

```diff
@@ -1624,6 +1653,274 @@ def energy_losses(
+        return losses / self.capacity(per_capacity)
+
+    def technical_loss_ratio(self) -> float:
+        """Calculate technical losses based on the project type.
+
+        This method is adopted from ORCA where a 1% hysterisis loss is applied for fixed-bottom
+        turbines. For floating turbines, this is 1% hysterisis, 0.1% for onboard equipment, and 0.1%
+        for rotor misalignment (0.01197901 total).
+
+        Returns
+        -------
+        float
+            The technical loss ratio.
+        """
+        if self.turbine_type == "floating":
+            return 1 - (1 - 0.01) * (1 - 0.001) * (1 - 0.001)
+        else:
+            return 0.01
+
+    def electrical_loss_ratio(self) -> float:
+        """Calculate electrical losses based on ORBIT parameters.
+
+        Returns
+        -------
+        float
+            The electrical loss ratio.
+
+        """
+        depth = self.orbit_config_dict["site"]["depth"]
+        distance_to_landfall = self.orbit_config_dict["site"]["distance_to_landfall"]
+        # ORCA formula
+        electrical_loss_ratio = (
+            2.20224112
+            + 0.000604121 * depth
+            + 0.0407303367321603 * distance_to_landfall
+            + -0.0003712532582 * distance_to_landfall**2
+            + 0.0000016525338 * distance_to_landfall**3
+            + -0.000000003547544 * distance_to_landfall**4
+            + 0.0000000000029271 * distance_to_landfall**5
+        ) / 100
+
+        return electrical_loss_ratio
+
+    def total_loss_ratio(
+        self,
+        environmental_loss_ratio: float | None = None,
+    ) -> float:
+        """Calculate total losses based on environmental, availability, wake, technical, and
+        electrical losses.
+
+        Parameters
+        ----------
+        environmental_loss_ratio : float, optional
+            The decimal environmental loss ratio to apply to the energy production. If None, then
+            it will attempt to use the :py:attr:`environmental_loss_ratio` provided in the Project
+            configuration. Defaults to 0.0159.
+
+        Returns
+        -------
+        float
+            The total loss ratio.
+
+        """
+        # Check that the environmental loos ratio exists
+        if environmental_loss_ratio is None:
+            if (environmental_loss_ratio := self.environmental_loss_ratio) is None:
+                raise ValueError(
+                    "`environmental_loss_ratio` wasn neither defined in the Project settings nor"
+                    " provided in the keyword arguments."
+                )
+
+        total_loss_ratio = 1 - (
+            (1 - environmental_loss_ratio)
+            * (1 - (1 - self.availability(which="energy", frequency="project", by="windfarm")))
+            * (1 - self.wake_losses(ratio=True))
+            * (1 - self.technical_loss_ratio())
+            * (1 - self.electrical_loss_ratio())
+        )
+
+        return total_loss_ratio
+
+    def losses_breakdown(
```

Comment:
git pushThis should really be incorporated into the `energy_losses` method using a new keyword argument called `breakdown` because the loss amounts may need to be broken down across time as well.